### ISSUE TITLE
[1.12] Resizing the window removes warning info

### ISSUE THREAD

**Issue by Darkhax** [SlimeKnights/TinkersConstruct#3102]
If a mod cancels the event, the warning/error message given will be replaced with the normal info when the screen is resized. 

![animation](https://user-images.githubusercontent.com/2250798/29300866-8f617c20-8135-11e7-9067-b3fe5534f661.gif)

Code used
```java
    @SubscribeEvent
    public void onToolCrafted(ToolCraftingEvent event) {
    	
    	System.out.println("Tool Craft - " + event.getItemStack().toString());
    	event.setCanceled("NO TOOL FOR YOU!");
    }
```
TCon Version: 1.12-2.7.2.508
Mantle: 1.3.1.19

**Edit 1:** Changing the tool type without changing the parts, will also cause the info to update incorrectly. 

**Edit 2:** When opening the table, if it already has the parts needed for a tool inside, things will start acting weirdly. For example, the event for when a tool is crafted will be fired with a null player, and it can cause some issues if the event listener doesn't detect this case and cancel earlier. Cancelling the message in this early state will not apply the error message.

**Comment by darkhax** [comment]
As an addendum to this issue, doing something like switching the tool type will also reset the info.

**Comment by bonii-xx** [comment]
Yeah, an update is missing somewhere. Weird thing is that it should always run through the same code.

**Comment by KnightMiner** [comment]
Pretty sure this is fixed in 1.16. The fix was a bit too complicated for me to want to port it to 1.12, so going to close this as won't fix.
